theano/compile/function_module.py

```diff
@@ -485,16 +486,178 @@ def __setitem__(self, item, value):
         self.value[item] = value
 
     def __copy__(self):
-        defaults = [default for _1, _2, default in self.defaults]
-        cpy = self.maker.create(defaults, trustme=True)
-        for (input, _1, _2), here, there in zip(self.indices,
-                                                self.input_storage,
-                                                cpy.input_storage):
-            if input.mutable and here is not None:
-                there.data = copy.copy(here.data)
+        """
+        Copy a function. Copied function have separate intermediate
+        storages and output storages with original function
+        """
+        return self.copy()
+
+    def copy(self, share_memory=False, swap=None, delete_updates=False):
+        """
+        Copy this function. Copied function will have separated maker and
+        fgraph with original function. User can choose whether to separate
+        storage by changing the share_memory arguments.
+        Note:
+        We reuse original Out instances but In instances, therefore variables
+        in Ins and Outs are not in the same graph. Please avoid using variables
+        is Ins and Outs to reinitialize another Function or FunctionGraoh.
+        ---------------------
+        Params:
+            share_memory -- { boolean } Default is False. When True, two
+            function share intermediate storages(storages except input and
+            output storages). Otherwise two function will only share partial
```

Comment:
"two function" two indicate plurials. But I'm not a native english. I think it should be

Otherwise the two functions ...

If you don't agree, I won't insist as I could be wrong.